controller/readonly.go

```diff
@@ -10,25 +13,106 @@ var _ Controller = (*ReadOnly)(nil)
 
 // ReadOnly is the controller to run in read-only mode
 type ReadOnly struct {
-	conf *config.Config
+	*baseController
 }
 
 // NewReadOnly configures and initializes a new ReadOnly controller
-func NewReadOnly(conf *config.Config) *ReadOnly {
-	return &ReadOnly{
-		conf: conf,
+func NewReadOnly(conf *config.Config) (Controller, error) {
+	// Run the driver with logging to output the Terraform plan to stdout
+	if tfConfig := conf.Driver.Terraform; tfConfig != nil {
+		tfConfig.Log = config.Bool(true)
```

Comment:
This is specifically for the Terraform driver, it might look different for other drives when we're at that point.

As of right now, tfConfig is never nil because of [config validation ensuring that the Terraform Driver is configured](https://github.com/hashicorp/consul-terraform-sync/blob/master/config/driver.go#L84) (if unset it would have been configured with defaults in `config.Finalize()`)

Reply:
ah gotcha! thank you :)